includes/wcpdf-functions.php

```diff
@@ -721,3 +721,190 @@ function wpo_wcpdf_get_multilingual_languages(): array {
 	return apply_filters( 'wpo_wcpdf_multilingual_languages', $languages );
 }
 
+/**
+ * Get image mime type
+ *
+ * @param string $src
+ * @return string
+ */
+function wpo_wcpdf_get_image_mime_type( string $src ): string {
+	$mime_type = '';
+
+	if ( empty( $src ) ) {
+		return $mime_type;
+	}
+
+	// Check if 'getimagesize' function exists and try to get mime type for local files
+	if ( function_exists( 'getimagesize' ) ) {
+		$image_info = @getimagesize( $src );
+
+		if ( $image_info && isset( $image_info['mime'] ) ) {
+			$mime_type = $image_info['mime'];
+		} else {
+			wcpdf_log_error( 'getimagesize failed for file: ' . $src );
+		}
+	}
+
+	// Fallback to 'finfo_file' if mime type is empty for local files only (no remote files allowed)
+	if ( empty( $mime_type ) && function_exists( 'finfo_open' ) && ! filter_var( $src, FILTER_VALIDATE_URL ) ) {
+		$finfo = finfo_open( FILEINFO_MIME_TYPE );
+
+		if ( $finfo ) {
+			$mime_type = finfo_file( $finfo, $src );
+			finfo_close( $finfo );
+		} else {
+			wcpdf_log_error( 'Fileinfo failed to open for file: ' . $src );
+		}
+	}
+
+	// Handle remote files
+	if ( empty( $mime_type ) && filter_var( $src, FILTER_VALIDATE_URL ) ) {
+		$context = stream_context_create( array(
+			'http' => array(
+				'method'        => 'HEAD',
+				'ignore_errors' => true,
+			),
+			'https' => array(
+				'method'           => 'HEAD',
+				'ignore_errors'    => true,
+				'verify_peer'      => false,
+				'verify_peer_name' => false,
+			),
+		) );
+
+		$headers = @get_headers( $src, 1, $context );
+
+		if ( $headers ) {
+			if ( isset( $headers['Content-Type'] ) ) {
+				$mime_type = is_array( $headers['Content-Type'] ) ? $headers['Content-Type'][0] : $headers['Content-Type'];
+			} else {
+				wcpdf_log_error( 'Content-Type header not found for URL: ' . $src );
+			}
+		} else {
+			wcpdf_log_error( 'Failed to get headers for URL: ' . $src );
+		}
+	}
+
+	// Determine using WP functions
+	if ( empty( $mime_type ) ) {
+		$path      = wp_parse_url( $src, PHP_URL_PATH );
+		$file_info = wp_check_filetype( $path );
+		$mime_type = $file_info['type'] ?? 'application/octet-stream';
```

Comment:
I was testing a random image URL with this approach. The URL is served through a CDN and uses query parameters for dynamic content delivery:


`$src = 'https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcRrTFrhr_-pYR74jUgOy7IerAoHAX3zPIZZcg&s'`

I got these results:

```php
$path = wp_parse_url( $src, PHP_URL_PATH ); // result: "/images"
$file_info = wp_check_filetype( $path ); // result: [  "ext" => false,  "type" => false,]
```

Reply:
Should be fixed now.

Reply:
The new approach is working correctly.